crates/evm-utils/src/evm_runner.rs

```diff
@@ -124,18 +124,46 @@ impl EvmRunner {
     ) -> Result<SuccessResult, EvmRunnerError> {
         let mut evm = Evm::builder()
             .with_db(&mut self.db)
-            .modify_tx_env(|tx| {
-                tx.caller = caller.unwrap_or(address!("0000000000000000000000000000000000000000"));
-                tx.gas_limit = u64::MAX;
-                tx.transact_to = TxKind::Call(address);
-                tx.data = calldata.into();
-                tx.value = U256::from(value.unwrap_or(U256::ZERO));
-                tx.chain_id = Some(1);
-            })
+            .modify_tx_env(|tx| Self::set_tx_env_for_call(address, calldata, caller, value, tx))
             .build();
 
-        let result = evm.transact_commit()?;
+        Self::handle_call_result(evm.transact_commit()?)
+    }
+
+    /// Apply `call` transaction to given `address` with `calldata`, but do not commit changes
+    /// to db. Return a tuple of `gas_used` and `return_data`.
+    pub fn call_ref(
```

Comment:
Maybe a better name:

```suggestion
    pub fn dry_call(
```

Reply:
done in https://github.com/Cardinal-Cryptography/zkOS-monorepo/pull/130/files